tests/test_dom.py

```diff
@@ -115,6 +115,53 @@ class E(D):
     assert e_css[1][1:] == ("C", -2)
 
 
+def test_component_classes_inheritance():
+    """Test if component classes are inherited properly."""
+
+    class A(DOMNode):
+        COMPONENT_CLASSES = {"a-1", "a-2"}
+
+    class B(A):
+        COMPONENT_CLASSES = {"b-1"}
+        _inherit_component_classes = False
```

Comment:
Given that this will be useful for people using the library, wanting to inherit from a widget and decide if they want the component classes or not, does it make sense to have the inherit control switch be a "private" value?

Reply:
Thanks for pointing this out! The problem is that I misunderstood how `_inherit_component_classes` should be set.